File: app/src/main/java/com/getman/varnabeach/lifecycle/BeachListViewModel.java
package com.getman.varnabeach.lifecycle;

import android.app.Application;

import androidx.lifecycle.AndroidViewModel;
import androidx.lifecycle.LiveData;

import com.getman.varnabeach.data.Repository;
import com.getman.varnabeach.room.Beach;

import java.util.List;

public class BeachListViewModel extends AndroidViewModel {
    private final LiveData<List<Beach>> allBeaches;

    public BeachListViewModel(Application application) {
        super(application);

        Repository repository = Repository.getInstance(application);
        allBeaches = repository.getAllBeaches();
    }

    public LiveData<List<Beach>> getAllBeaches() {
        return allBeaches;
    }
}

File: app/src/main/java/com/getman/varnabeach/lifecycle/BeachListViewModel.kt
package com.getman.varnabeach.lifecycle

import androidx.lifecycle.LiveData
import androidx.lifecycle.MutableLiveData
import androidx.lifecycle.ViewModel
import com.getman.varnabeach.data.Repository
import com.getman.varnabeach.room.Beach
import dagger.hilt.android.lifecycle.HiltViewModel
import javax.inject.Inject

@HiltViewModel
class BeachListViewModel @Inject constructor(
    repository: Repository
) : ViewModel() {
    val allBeaches: LiveData<List<Beach>> = repository.allBeaches
    private val selectedBeachMutable = MutableLiveData<Beach>()
    val selectedBeach: LiveData<Beach> get() = selectedBeachMutable

    fun select(selectedBeach: Beach) {
        selectedBeachMutable.value = selectedBeach
    }
}